.github/actions/download-artifact-retry/action.yaml

```diff
@@ -0,0 +1,31 @@
+name: Download artifact with retry
```

Comment:
Regarding the job file name: I'd suggest `.github/actions/download-artifact-with-retry/action.yaml` instead of `.github/actions/download-artifact-retry/action.yaml`. Although you'd have to rename `upload-artifact-retry` to `upload-artifact-with-retry` for consistency.